web/bfex/components/data_pipeline/tasks/googlescholar.py

```diff
@@ -34,31 +28,31 @@ def run(self, data):
         :return: last faculty member handled
         """
         
-        no_text_count = 0
-        for faculty in data:
+        faculty = data
+        if isinstance(faculty, str):
+            faculty_name = faculty
+        else:
             faculty_name = faculty.name
-    
-            search_results = Faculty.search().query('match', name=faculty_name).execute()
-            if len(search_results) > 1:
-                # Shouldn't happen, but could.
-                raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
-
-            search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="GoogleScholar")
-            search_dup.delete()
 
-            faculty = search_results[0]
-            if faculty.google_scholar is not None and "http" in faculty.google_scholar:
-                scraper = ScraperFactory.create_scraper(faculty.google_scholar, ScraperType.GOOGLESCHOLAR)
-                scrapps = scraper.get_scrapps()
-                for scrapp in scrapps:
-                    doc = Document()
-                    doc.source = "GoogleScholar"
-                    doc.faculty_id = faculty.faculty_id
-                    doc.text = scrapp.title
-                    doc.save()
-            else:
-                no_text_count += 1
-        print("NO TEXT COUNT = ", no_text_count)
+        search_results = Faculty.search().query('match', name=faculty_name).execute()
+        if len(search_results) > 1:
+            # Shouldn't happen, but could.
+            raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
+
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="GoogleScholar")
+        search_dup.delete()
+
+        faculty = search_results[0]
+        if faculty.google_scholar is not None and "http" in faculty.google_scholar:
+            scraper = ScraperFactory.create_scraper(faculty.google_scholar, ScraperType.GOOGLESCHOLAR)
+            scrapps = scraper.get_scrapps()
+            for scrapp in scrapps:
+                doc = Document()
+                doc.source = "GoogleScholar"
```

Comment:
I thought this was going to be a tuple with a string value? :P

Reply:
yeah we messed the doc up